src/NuGet.Core/NuGet.Frameworks/NuGetFramework.cs

```diff
@@ -57,8 +60,47 @@ public NuGetFramework(string frameworkIdentifier, Version frameworkVersion, stri
 
             _frameworkIdentifier = frameworkIdentifier;
             _frameworkVersion = NormalizeVersion(frameworkVersion);
-            _frameworkProfile = frameworkProfile ?? string.Empty;
+
+            IsNet5Era = (_frameworkVersion.Major >= Version5 && StringComparer.OrdinalIgnoreCase.Equals(FrameworkConstants.FrameworkIdentifiers.NetCoreApp, _frameworkIdentifier));
+
+            _frameworkProfile = profile ?? string.Empty;
+            Platform = string.Empty;
+            PlatformVersion = FrameworkConstants.EmptyVersion;
+        }
+
+        /// <summary>
+        /// Creates a new NuGetFramework instance, with an optional platform and platformVersion (only available for net5.0+)
+        /// </summary>
+        public NuGetFramework(string frameworkIdentifier, Version frameworkVersion, string platform, Version platformVersion)
+        {
+            if (frameworkIdentifier == null)
+            {
+                throw new ArgumentNullException("frameworkIdentifier");
+            }
+
+            if (frameworkVersion == null)
+            {
+                throw new ArgumentNullException("frameworkVersion");
+            }
+
+            if (platform == null)
+            {
+                throw new ArgumentNullException("platform");
+            }
+
+            if (platformVersion == null)
+            {
+                throw new ArgumentNullException("platformVersion");
+            }
+
+            _frameworkIdentifier = frameworkIdentifier;
+            _frameworkVersion = NormalizeVersion(frameworkVersion);
+            _frameworkProfile = string.Empty;
+
             IsNet5Era = (_frameworkVersion.Major >= Version5 && StringComparer.OrdinalIgnoreCase.Equals(FrameworkConstants.FrameworkIdentifiers.NetCoreApp, _frameworkIdentifier));
+
+            Platform = IsNet5Era ? platform : string.Empty;
+            PlatformVersion = IsNet5Era ? NormalizeVersion(platformVersion) : FrameworkConstants.EmptyVersion;
```

Comment:
Should we verify that `PlatformVersion` is not `EmptyVersion` when `Platform` is not `string.Empty`?  Only projects are allowed to omit platform version, but that means that it uses the SDK's default platform version, not that platform version is empty.

Also, should we verify that `PlatformVersion` is `EmptyVersion` when `Platform` is `string.Empty`?  It doesn't make sense to not have a platform, but still to have a platform version.